test/test_charm.py

```diff
@@ -561,18 +561,52 @@ def test_containers_storage(self):
   other:
     type: filesystem
     location: /test/other
+    properties:
+      - transient
 containers:
   test1:
     mounts:
       - storage: data
         location: /test/storagemount
       - storage: other
         location: /test/otherdata
+    resource: ubuntu-22.10
+  test2:
+    bases:
+      - name: ubuntu
+        channel: '23.10'
+        architectures:
+         - amd64
+      - name: ubuntu
+        channel: 23.04/stable/fips
+        architectures:
+         - arm
 """)
         self.assertIsInstance(meta.containers['test1'], ops.ContainerMeta)
         self.assertIsInstance(meta.containers['test1'].mounts["data"], ops.ContainerStorageMeta)
         self.assertEqual(meta.containers['test1'].mounts["data"].location, '/test/storagemount')
         self.assertEqual(meta.containers['test1'].mounts["other"].location, '/test/otherdata')
+        self.assertEqual(meta.storages['other'].properties, ['transient'])
+        self.assertEqual(meta.containers['test1'].resource, 'ubuntu-22.10')
+        assert meta.containers['test2'].bases is not None
+        self.assertEqual(len(meta.containers['test2'].bases), 2)
+        self.assertEqual(meta.containers['test2'].bases[0].os_name, 'ubuntu')
+        self.assertEqual(meta.containers['test2'].bases[0].channel, '23.10')
+        self.assertEqual(meta.containers['test2'].bases[0].architectures, ['amd64'])
+        self.assertEqual(meta.containers['test2'].bases[1].os_name, 'ubuntu')
+        self.assertEqual(meta.containers['test2'].bases[1].channel, '23.04/stable/fips')
+        self.assertEqual(meta.containers['test2'].bases[1].architectures, ['arm'])
+        with self.assertRaises(ModelError):
```

Comment:
Can you please add a comment above the `with` line about why this is failing? It's not immediately obvious to me.